### PR TITLE
refactor: documentation and test coverage for ref_split

### PR DESCRIPTION
**Pull request by soofstad**
## What does this pull request change?
- Fixes some unresolved comments from #429 

## Why is this pull request needed?
- better documentation and test coverage

## Issues related to this change:
none

### CODE REVIEW

File: src/services/document_service.py [hunk 1]
```diff
@@ -137,6 +137,19 @@ def save(
...
+        node: The Node to save
+        path: A Filesystem equivalent path for this node. Used when writing zip-files.
+        combined_document_meta:  The combined meta information.
+            For example:
+                nodeA
+                    nodeB
+                        nodeC
+            Here, combined_document_meta is the combined _meta_ information of node A, B and C.
+            (this meta info can be found with _collect_entity_meta_by_path() util function).-
+        initial:  When true. The function will move up the tree until it finds a storage_contained node, and start saving from there.
```
Comment (ingeridhellen) [inline src/services/document_service.py:150]:
```suggestion
        initial:  When true, the function will move up the tree until it finds a node that is storage non-contained, and start saving from there.
```
right?

**Review by ingeridhellen — APPROVED** [review]
Great! Love the documentation

**Review by KristianKjerstad — APPROVED** [review]

File: src/services/document_service.py (hunk 1 above)
Comment (soofstad) [reply src/services/document_service.py:150]: right :)

File: src/common/utils/resolve_reference.py [hunk 2]
```diff
@@ -24,8 +24,9 @@ def split_reference(reference: str) -> list[str]:
             continue
         if remaining_ref[0] in "[(":
             prefix_delim = remaining_ref[0]
-            content = re.split(r"[\[\]\(\)]", remaining_ref, 2)[1]
-            parts.append(f"{prefix_delim}{content}{']' if prefix_delim == '[' else ')'}")
+            closer = "]" if prefix_delim == "[" else ")"
```
[the remark this answers is not in the document]
Comment (soofstad) [reply src/common/utils/resolve_reference.py:27]: fixed :+1:

**Pull request merged**